### PR TITLE
Website: Tiny UI Improvements

### PR DESCRIPTION
**Pull request by thisisobate**
What do these changes do?
-------------------------

I noticed some flaky behaviors on the website.
This is a follow-up PR to fix all of them

- [x] nav list style is showing on mobile
- [x] mobile menu background is not placed properly- item below overshadows one on top
- [x] theme switch/toggler is broken on mobile

Related issue number
--------------------

Relates to #1449

### CODE REVIEW

**Comment by thisisobate** [comment]
cc: @sighingnow

**Review by sighingnow — APPROVED** [review]

**Comment by sighingnow** [comment]
Merged. Thanks!

File: docs/_static/css/index.css [hunk 1]
```diff
@@ -182,6 +182,11 @@ button {
   padding: 100px 3% 100px;
 }
 
+.hero-section {
+  position: relative;
+  z-index: -1;
+}
+
```
Comment (sighingnow) [inline docs/_static/css/index.css:189]: I had to revert this change as it caused the buttons "Get started" and "API Reference" in the main page unclickable.
